src/docs/backend/control-silo.mdx

```diff
@@ -0,0 +1,178 @@
+---
+title: "Control Silo"
+---
+
+Within the Control Silo are features that allow us to provide backwards compatibility for both customer API usage, and integrations
+
+# API Gateway
+
+Prior to our multi-region offering, all API traffic from customers would be sent to [sentry.io](http://sentry.io) . We need to continue supporting these URLs despite some customers residing in the `us` region and others in the `de`region. The API Gateway functionality of control silo will introspect requests and proxy requests to the appropriate region based on path parameters in the request. Requests can be routed via:
+
+- Organization slug or id
+- Sentry App Installation UUID
+- Sentry App Slug
+- DSN host
+- A static list of URL names (region pin list)
+
+<Alert level="info">
+💡 Requests going to [sentry.io](http://sentry.io) do not meet data residency requirements. Customers will need to use region domains in order to not have requests go to the US.
+</Alert>
+
+The API Gateway is implemented as a middleware that only takes action when Sentry is running as Control Silo. After matching URLs, the middleware checks the silo mode of the matched endpoint. If the matched endpoint is a **region** endpoint, the API Gateway attempts to route the request to a region based on a few heuristics.
+
+### Slug, id, and UUID based routing
+
+For requests that use `organization_slug`, `organization_id_or_slug`, `sentry_app_slug` or `uuid` parameters the API Gateway can resolve the organization’s region using data present in control silo. Once the region is resolved requests are synchronously proxied to the appropriate region.
+
+### DSN host
+
+The legacy error-embed page requires specialized handling, and we extract the
+destination from the `dsn` query string parameter.
+
+### Region Pin List
+
+We have a handful of API endpoints that lack useful hints to route requests between regions. When a request is made to one of these paths we assume the intent is to send a request to the `us` region.
+
+### Direct Location Header
+
+When a request is handled by the API Gateway, the response delivered to the client will contain `X-Sentry-Proxy-Url` header which contains the URL that clients should use to avoid the additional latency incurred by the API Gateway.
+
+# Integration Credential Proxy
+
+Integrations can be shared by multiple organizations. For example, a single MS teams workspace can be connected to multiple organizations, and those organizations can live in multiple regions. A few of our integrations leverage refresh tokens and use short lived access tokens. To avoid racy updates when tokens become invalidated and need to be refreshed, we send outbound requests to control silo, which acts as a proxy for the integration service. By funnelling all outbound traffic for integrations that require refresh tokens through control silo we can more easily co-ordinate token replacements.
+
+```mermaid
```

Comment:
Preview doesn't like to do this automatically, you'll probably need to do what we did here: https://github.com/getsentry/develop/blob/master/src/docs/architecture.mdx?plain=1#L9